db/db_options_test.cc

```diff
@@ -178,6 +178,22 @@ TEST_F(DBOptionsTest, EnableAutoCompactionAndTriggerStall) {
   }
 }
 
+TEST_F(DBOptionsTest, EnableAutoCompactionShouldTriggerCompaction) {
+  Options options;
+  options.create_if_missing = true;
+  options.level0_file_num_compaction_trigger = 1000;
+  Reopen(options);
+  for (int i = 0; i < 3; i++) {
+    ASSERT_OK(Put(ToString(i), ToString(i)));
```

Comment:
Maybe we can write 3 files with overlapping ranges to disallow trivial move, this can be done by writing the same keys/values in all the 3 files